test/AutoDiff/compiler_crashers/tf922-adjoint-value-type-mismatch.swift

```diff
@@ -0,0 +1,23 @@
+// RUN: not --crash %target-swift-emit-sil %s
+// REQUIRES: asserts
+
+// TF-922: Adjoint value type mismatch assertion failure during direct adjoint
+// accumulation in differentiation transform pullback generation.
+
+@differentiable
+func TF_922(_ x: Float) -> [Float] {
+  var result: [Float] = [x]
+  let result2 = true ? result : result
+  let result3 = true ? result : result2
+  return result3
+}
+
+// Assertion failed: (lhs->getType() == rhs->getType() && "Adjoints must have equal types!"), function accumulateDirect, file swift/lib/SILOptimizer/Mandatory/Differentiation.cpp, line 7654.
```

Comment:
Let's add some debug log before the assertion in `accumulateDirect` to print `lhs` and `rhs`.